pkgs/build-support/rust/default.nix

```diff
@@ -199,7 +199,7 @@ stdenv.mkDerivation (args // {
       -executable ! \( -regex ".*\.\(so.[0-9.]+\|so\|a\|dylib\)" \))
   '';
 
-  checkPhase = args.checkPhase or (let
+  installCheckPhase = args.checkPhase or (let
```

Comment:
If `checkPhase` is changed to `installCheckPhase`, should `doCheck`—

https://github.com/NixOS/nixpkgs/blob/deb78151a9438fb202842d9dbe348365cfd767e0/pkgs/build-support/rust/default.nix#L213

—be changed to `doInstallCheck`?  By my reading of it, [section 6.5.9 of the Nixpkgs manual](https://nixos.org/nixpkgs/manual/#ssec-installCheck-phase) seems to say that `installCheckPhase` won't be run unless `doInstallCheck = true` is set, as [section 6.5.6](https://nixos.org/nixpkgs/manual/#ssec-check-phase) seems to say for `checkPhase` and `doCheck`.

Reply:
Yes, with this it seems we have correct flags but we aren't running the checks.

However running the checks results in:
```
Running cargo test --release --target x86_64-apple-darwin --frozen --
    Blocking waiting for file lock on build directory
```
I don't have more time to look at this right now, I can come back to it tomorrow.

I won't bother reverting this as having correct flags without checks seems preferable over the status quo.

Reply:
This seems to break the `rls` build on master:

```
checking for references to /build/ in /nix/store/fa4hvsckbjq80dhck6lxmwgx3waxjck1-rls-1.44.1...
running install tests
/build/rustc-1.44.1-src/src/tools/rls /build/rustc-1.44.1-src
Running cargo test --release --target x86_64-unknown-linux-gnu --frozen --
    Blocking waiting for file lock on build directory
```

Reply:
It is wrong. It should be at least `installCheckPhase = args.installCheckPhase or ...`.
Now it overrides the original `installCheckPhase` and also breaks #91359 (infinite `Blocking waiting for file lock`).

Reply:
I'm trying to fix the original problem in a different way atm to get this patch reverted btw.

Reply:
Thanks for trying to fix this in a better way @Ma27.
Sorry for breaking packages, I should have tested more with nixpkgs-review.